corehq/apps/hqwebapp/static/hqwebapp/js/select2_handler_v4.js

```diff
@@ -0,0 +1,119 @@
+        self.multiple = options.multiple || false;
+        self.value = ko.observable();
+
+        self.clear = function () {
+            var fieldInput = self.utils.getField();
+            fieldInput.select2('val', '');
+        };
+
+        self.getHandlerSlug = function () {
+            // This is the slug for the AsyncHandler you'll be using on the server side
+            throw new Error('getHandlerSlug must be implemented');
+        };
+
+        self.getExtraData = function () {
+            return {};
+        };
+
+        self.processResults = function (response) {
+            // override this if you want to do something special with the response.
+            return response;
+        };
+
+        self.createNewChoice = function (term, selectedData) {
+            // override this if you want the search to return the option of creating whatever
+            // the user entered.
+        };
+
+        self.formatResult = function (result) {
+            return result.text;
+        };
+
+        self.formatSelection = function (result) {
+            return result.text;
+        };
+
+        self.getInitialData = function (element) {
+            // override this if you want to format the value that is initially stored in the field for this widget.
+        };
+
+        self.utils = {
+            getField: function () {
+                return $('[name="' + self.fieldName + '"]');
+            },
+        };
+
+        self.init = function () {
+            var fieldInput = self.utils.getField();
+
+
+            fieldInput.select2(_.extend({
+                multiple: true,
+                tags: fieldInput.data("choices"),
+            }, {}));
+
+            fieldInput.prepend('<option value="-1"></option>').select2({
+                minimumInputLength: 0,
+                allowClear: true,
+                multiple: self.multiple,
+                placeholder: {
+                    id: '-1',
+                    text: self.placeholder,
+                },
+                ajax: {
+                    quietMillis: 150,
+                    url: '',
+                    dataType: 'json',
+                    type: 'post',
+                    data: function (term) {
+                        var data = self.getExtraData(term);
+                        data['handler'] = self.getHandlerSlug();
+                        data['action'] = self.fieldName;
+                        data['searchString'] = term;
+                        return data;
+                    },
+                    results: self.processResults,
+                    500: function () {
+                        self.error(
+                            gettext("There was an issue communicating with the server. Please try back later.")
+                        );
+                    },
+                },
+                createSearchChoice: self.createNewChoice,
+                formatResult: self.formatResult,
```

Comment:
`formatResult` and `formatSelection` are deprecated in v4, see https://select2.org/upgrading/migrating-from-35#renamed-templating-options